claims_hosp/delphi_claims_hosp/update_indicator.py

```diff
@@ -133,14 +133,76 @@ def geo_reindex(self, data):
         data_frame.fillna(0, inplace=True)
         return data_frame
 
-    def update_indicator(self, input_filepath, outpath, logger):
+    def update_indicator_to_df(self, input_filepath, logger):
         """
         Generate and output indicator values.
 
         Args:
             input_filepath: path to the aggregated claims data
-            outpath: output path for the csv results
+        """
+        self.shift_dates()
+        final_output_inds = (self.burn_in_dates >= self.startdate) & (self.burn_in_dates <= self.enddate)
+
+        # load data
+        base_geo = Config.HRR_COL if self.geo == Config.HRR_COL else Config.FIPS_COL
+        data = load_data(input_filepath, self.dropdate, base_geo)
+        data_frame = self.geo_reindex(data)
 
+        # handle if we need to adjust by weekday
+        wd_params = (
+            Weekday.get_params_legacy(
+                data_frame,
+                "den",
+                ["num"],
+                Config.DATE_COL,
+                [1, 1e5],
+                logger,
+            )
+            if self.weekday
+            else None
+        )
+        output_df = pd.DataFrame()
+        if not self.parallel:
+            for geo_id, sub_data in data_frame.groupby(level=0):
+                sub_data.reset_index(inplace=True)
+                if self.weekday:
+                    sub_data = Weekday.calc_adjustment(wd_params, sub_data, ["num"], Config.DATE_COL)
+                sub_data.set_index(Config.DATE_COL, inplace=True)
+                res = ClaimsHospIndicator.fit(sub_data, self.burnindate, geo_id)
+                output_df = output_df.append(pd.DataFrame(res))
```

Comment:
suggestion (non-blocking): the [append method is deprecated](https://pandas.pydata.org/docs/whatsnew/v1.4.0.html#deprecated-dataframe-append-and-series-append) in favor of `pd.concat(list_of_dataframes)` in pandas 2 (for various reasons, performance being one), so if we instead build a list of dataframes and do a single concat after the for-loop, we can future proof this code for our migration to pandas 2.